testsupport/init.go

```diff
@@ -62,11 +67,71 @@ func waitForOperators(t *testing.T) {
 	kubeconfig, err := util.BuildKubernetesRESTConfig(*apiConfig)
 	require.NoError(t, err)
 
+	// creating another config which is used for creating resclient only,
+	//so that the main kubeconfig is not altered
+	restkubeconfig, err := util.BuildKubernetesRESTConfig(*apiConfig)
+	require.NoError(t, err)
+
 	cl, err := client.New(kubeconfig, client.Options{
 		Scheme: schemeWithAllAPIs(t),
 	})
 	require.NoError(t, err)
 
+	// Check if there is already a service account present for e2e test
+	sa := &corev1.ServiceAccount{}
+	err = cl.Get(context.TODO(), types.NamespacedName{Namespace: hostNs, Name: "e2e-test"}, sa)
+
+	// If not found proceed to create the e2e service account and the cluster role binding
+	if errors.IsNotFound(err) {
+		fmt.Printf("No Service Account for e2e test found, proceeding to create it")
+		err := cl.Create(context.TODO(), &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "e2e-test",
+				Namespace: hostNs}})
+		require.NoError(t, err, "Error in creating Service account for e2e test")
+
+		crb := rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "e2e-test-cluster-admin",
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     "cluster-admin",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      "e2e-test",
+					Namespace: hostNs,
+				},
+			},
+		}
+		fmt.Printf("Proceeding to create Cluster Role Binding for the Service Account")
+		err = cl.Create(context.TODO(), &crb)
+		require.NoError(t, err, "Error in Creating Cluster role binding")
+	} else if err != nil {
+		require.NoError(t, err, "Error fetching service accounts")
+	}
+
+	//upating the restkubeconfig ,which requires groupversion to create restclient
+	restkubeconfig.ContentConfig =
+		rest.ContentConfig{
+			GroupVersion:         &authv1.SchemeGroupVersion,
+			NegotiatedSerializer: scheme.Codecs,
+		}
+
+	//Creating a Restclient to be used in creation and checking of bearer token required for authentication
+	rclient, err := rest.RESTClientFor(restkubeconfig)
+	require.NoError(t, err, "Error in creating restclient")
+
+	//Creating a bearer token to be used for authentication(which is valid for 24 hrs)
+	bt, err := toolchaincommon.CreateTokenRequest(context.TODO(), rclient, types.NamespacedName{Namespace: hostNs, Name: "e2e-test"}, 86400)
+	require.NoError(t, err, "Error in creating Token")
```

Comment:
could you please extract this new code into a separate function - for example 
```
func getE2EServiceAccountToken() string
```
so it's easier to read?

Reply:
yep.. done